### PR TITLE
Add overwrite option to cfg.file and getEnv function

### PR DESCRIPTION
**Pull request by rohit-smpx**
- Added option to overwrite config. This was needed because when using cfg with global clis (https://github.com/smartprix/pg_backup) it will look in cwd for config.js and might find some other app's config file. So we need to overwrite before using it.
- getEnv: we keep doing `process.env.NODE_ENV || 'development'` everywhere so thought why not add it:
- Used test folder specific eslintrc to add mocha globals

### CODE REVIEW

**Comment by rohit-smpx** [comment]
Also fixed #20

File: src/cfg.js [hunk 1]
```diff
@@ -214,20 +222,29 @@ cfg.read = function (key) {
 	return fileCache[key];
 };
 
+const env = process.env.NODE_ENV || 'development';
```
Comment (hit12369) [inline src/cfg.js]: If we do this then changing `process.env.NODE_ENV` from the script would have no effect.

File: src/cfg.js [hunk 2]
```diff
@@ -214,20 +222,29 @@ cfg.read = function (key) {
...
+/**
+ * @memberof config
+ * @return {string}
+ */
+cfg.getEnv = function () {
```
Comment (hit12369) [inline src/cfg.js]: This should be aliased as `cfg.env()` too.

File: src/cfg.js (hunk 1 above)
Comment (hit12369) [reply src/cfg.js]: Maybe we should use `cfg.env()` everywhere instead of env.

**Comment by rohit-smpx** [comment]
@hit12369 Updated with changes. Also updated docs.

**Pull request merged**